emukit/core/discrete_parameter.py

```diff
@@ -0,0 +1,32 @@
+from typing import Iterable
+
+
+class DiscreteParameter(object):
+    """
+    A parameter that takes a discrete set of values where the order is important
+    """
+    def __init__(self, name: str, domain: Iterable):
+        """
+
+        :param name: Name of parameter
+        :param domain: valid values the parameter can have
+        """
+        self.name = name
+        self.domain = domain
+
+    def check_in_domain(self, x: Iterable) -> bool:
+        """
+        Checks if the points in x are in the set of allowed values
+
+        :param x: 1d numpy array of points to check
+        :return: A 1d numpy array which contains a boolean indicating whether each point is in domain
```

Comment:
type annotations in function call and docstring do not match

Reply:
Good point, I'll change the docstring, the type annotation is correct.